src/java/com/palantir/cassandra/utils/RangeTombstoneCounter.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.cassandra.utils;
+
+import java.util.Iterator;
+
+import org.apache.cassandra.db.DeletionInfo;
+import org.apache.cassandra.db.RangeTombstone;
+
+public class RangeTombstoneCounter
+{
+    private final int droppableRangeTombstones;
+    private final int rangeTombstones;
+
+    public static final RangeTombstoneCounter EMPTY = new RangeTombstoneCounter(0, 0);
+
+    public RangeTombstoneCounter(int droppableRangeTombstones, int rangeTombstones) {
+        this.droppableRangeTombstones = droppableRangeTombstones;
+        this.rangeTombstones = rangeTombstones;
+    }
+
+    public static RangeTombstoneCounter create(int gcBefore, DeletionInfo deletionInfo) {
```

Comment:
Are we positive that this works?  I would expect that after exhausting the `container.deletionInfo()` iterator any further iterations would have no results.

I'm also generally nervous about this approach.  Other iterators "wrap" because it's very expensive - O(n) - to iterate the cells more than once.

I think you'll need to follow the model of the `CountingCellIterator` (where you use a delegator but also add additional functionality), and then wrap the location where the tombstone iterator is _actually_ called.

Reply:
Yup, iterator() is immutable as seen here: https://github.com/palantir/cassandra/blob/palantir-cassandra-2.2.18/src/java/org/apache/cassandra/db/RangeTombstoneList.java#L367

We do already perform an O(N) (worst-case) here: https://github.com/palantir/cassandra/blob/palantir-cassandra-2.2.18/src/java/org/apache/cassandra/db/RangeTombstoneList.java#L902

However, I think it sounds like we'd rather have something "more invasive" than computational costs, so I'll get back to the drawing board.

Reply:
How many times are we doing N though?  I'm unconfident about wanting to make it 2 or 3 x N

Reply:
Yeah, it's at least 2N :(

Reply:
The other PR is here: https://github.com/palantir/cassandra/pull/233